src/services/todo-service.ts

```diff
@@ -1,11 +1,27 @@
 import model, { ToDo } from "../models/todo-model";
+import { OrderType } from "../types/order-type";
 
 export const todoService = {
   insert: async (toDo: ToDo): Promise<ToDo | void> => {
     return await model.create(toDo);
   },
 
-  listAll: async (): Promise<ToDo[] | void> => {
+  listAll: async (
+    sortBy?: keyof ToDo,
+    order?: OrderType,
+    searchString?: string
+  ): Promise<ToDo[] | void> => {
+    if (searchString) {
+      return await model.find({
+        $or: [
+          { title: { $regex: searchString } },
+          { text: { $regex: searchString } },
+        ],
+      });
+    }
+    if (sortBy && order) {
+      return await model.find().sort({ [sortBy]: order });
+    }
```

Comment:
What if we want to search and sort in the same time, please rewrite this to support search and sort simultaneously